### PR TITLE
Fix messaging

### PR DESCRIPTION
**Pull request by ptgott**
In gravitational/docs#238, we will add a linter that lints for incorrect usage of Teleport terms. This change fixes linter violations to anticipate this new linter.

### CODE REVIEW

File: docs/pages/access-controls/guides/hardware-key-support.mdx [hunk 1]
```diff
@@ -27,22 +27,22 @@ Additionally, this feature can be configured to require touch for every Teleport
   Supported (`tsh`, `tctl`, and Teleport Connect):
     
     - Standard Teleport API requests (`tsh ls`, `tctl create`, etc.)
-    - Server Access
-    - Database Access
+    - Teleport SSH Service
```
Comment (zmb3) [inline docs/pages/access-controls/guides/hardware-key-support.mdx:30]: This doesn't seem right to me. This docs page is attempting to say that hardware key support works for accessing SSH servers or databases. Why are we mentioning the specific service in the Teleport process here?

File: docs/pages/architecture/session-recording.mdx [hunk 2]
```diff
@@ -49,9 +49,9 @@ Windows, database, and Kubernetes sessions are always recorded at the host runni
 the Teleport service for this session type, since there is no Teleport software
 running on the "node" (desktop, database, or Kubernetes cluster).
 
-For this reason, Teleport Windows, Database, and Kubernetes Access treats `node`
-and `proxy` identically (perform asynchronous recording) and `node-sync` and
-`proxy-sync` identically (perform synchronous recording).
+For this reason, the Teleport Desktop, Database, and Kubernetes Services treat
+`node` and `proxy` identically (perform asynchronous recording) and `node-sync`
+and `proxy-sync` identically (perform synchronous recording).
```
Comment (zmb3) [inline docs/pages/architecture/session-recording.mdx:54]: Remove trailing whitespace.

File: docs/pages/connect-your-client/introduction.mdx [hunk 3]
```diff
@@ -310,8 +310,8 @@ either directly or through proxy tunnels.
   Learn more about it from [Using the tsh Command Line Tool](./tsh.mdx).
 
 - Teleport Connect is a graphic utility for connecting to resources through
-Teleport. You can initiate Server Access, connect to databases, and start
-Desktop Access sessions. See [Using Teleport Connect](./teleport-connect.mdx).
+  Teleport. You can connect to servers, databases, and desktops. See [Using
```
Comment (zmb3) [inline docs/pages/connect-your-client/introduction.mdx:313]: I don't know how this got here, but it's incorrect. You can not connect to desktops with Teleport Connect - only servers, databases, and Kube clusters.

File: docs/pages/database-access/architecture.mdx [hunk 4]
```diff
@@ -1,13 +1,14 @@
 ---
-title: Database Access Architecture
-description: How Teleport Database Access works.
+title: Database Service Architecture
+description: How Teleport the Database Service works.
```
Comment (zmb3) [inline docs/pages/database-access/architecture.mdx:3]: This page isn't describing the architecture of just the database service, it's describing the architecture of the entire database access feature.

If you don't want to say _Database Access_ anymore, that's fine, but I think we should find a title that makes it clear this  is about how Teleport provides access to databases, which involves not only the database service, but the auth service and proxy service.

File: docs/pages/desktop-access/active-directory-manual.mdx [hunk 5]
```diff
@@ -1,25 +1,25 @@
 ---
-title: Desktop Access with Active Directory (Manual)
+title: Register Windows Desktops with Active Directory (Manual)
```
Comment (zmb3) [inline docs/pages/desktop-access/active-directory-manual.mdx:2]: This guide isn't really about registering desktops. How about "Connecting to desktops" instead?

File: docs/pages/preview/upcoming-releases.mdx [hunk 6]
```diff
@@ -22,13 +22,13 @@ You can find the full list of fixes and features in the [GitHub milestone](https
 
 ### Teleport 9
 
-Teleport 9.0 will focus on bringing Desktop Access to GA with support for
-session recording, per-session MFA, and clipboard support. Database Access
-will add self-hosted Redis, auto-discovery for Redshift clusers, and
-auto-IAM configuration improvements.
+Teleport 9.0 will focus on bringing the Desktop Service to GA with support for
```
Comment (zmb3) [inline docs/pages/preview/upcoming-releases.mdx:25]: This change probably isn't necessary. Nobody should be going to the upcoming releases page anymore to find out whats in a release from 9 months ago.

**Review by zmb3 — COMMENTED** [review]
Mostly looks fine, but I can't help but feel like we've made the docs less clear in places because we're trying so hard not to say "Server Access"

If this is the future direction, do we plan to update the sidebar entries too? Seems odd if they still say things like "Database Access" after we've removed all other references to the term

**Comment by ptgott** [comment]
@zmb3 Thanks for the review! I'm going to tweak the linter so we don't sacrifice clarity. I think we can:

- Ignore capitalized instances of "Database Access", "Server Access", etc. in titles. Since titles capitalize significant terms anyway, readers aren't going to think these refer to separate products.
- Catch capitalized instances of "Database Access", "Server Access", etc. in the body text and page description while recommending using the lowercased version instead, e.g., "database access". This way we don't need to go through odd contortions to talk about accessing resources.

What do you think?

**Comment by ptgott** [comment]
@alexfornuto Checking to see if you have any thoughts on this PR. I think it would be great to be able to merge the [linter PR](https://github.com/gravitational/docs/pull/238) in `gravitational/docs` so we can standardize Teleport terms across the docs.

**Review by alexfornuto — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

**Pull request merged**